src/ImageSharp.Web/Caching/ICacheKey.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using Microsoft.AspNetCore.Http;
+using SixLabors.ImageSharp.Web.Commands;
+
+namespace SixLabors.ImageSharp.Web.Caching
+{
+    /// <summary>
+    /// Defines a contract that allows the creation of cache keys (used by <see cref="ICacheHash"/> to create hashed file names for storing cached images).
+    /// </summary>
+    public interface ICacheKey
+    {
+        /// <summary>
+        /// Creates the cache key based on the specified context and commands.
+        /// </summary>
+        /// <param name="context">The HTTP context.</param>
+        /// <param name="commands">The commands.</param>
+        /// <param name="caseSensitive">If set to <c>true</c> the cache key should be generated in a case sensitive manner.</param>
+        /// <returns>
+        /// The cache key.
+        /// </returns>
+        string Create(HttpContext context, CommandCollection commands, bool caseSensitive);
```

Comment:
I think `caseSensitive` should be an implementation detail of the concrete class, or an option at this level.

Reply:
We're discussing that here: https://github.com/SixLabors/ImageSharp.Web/pull/206#discussion_r794052974.